docs/book/src/simple-external-plugin-tutorial/api.go

```diff
@@ -0,0 +1,46 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
```

Comment:
Could we add here a Makefile with a target to build the plugin in the .bin/ dir ( we need ignore this dir as well )
Then, could we add an e2e test that just creates a project using the plugin and test it out against the Kind? 
That would be here: `kubebuilder/test/e2e/external-plugin/`

Reply:
I feel like this would be best in a follow up. I was hoping to keep the scope of this PR smaller (i.e just adding the sample) and add the e2e testing components in a later PR (I think @rashmigottipati mentioned that she had already started on some e2e testing work).

Reply:
That is fine.

Reply:
@camilamacedo86 I added a Makefile and a target named `build` that when run will build the sample external plugin and output the binary to `./bin/sampleexternalplugin` in commit https://github.com/kubernetes-sigs/kubebuilder/pull/2810/commits/ee04eee08658465719e07af06f30d100b7484bb1

If we still want to do a simple e2e test to ensure that the phase 2 plugins are working properly with this sample plugin I can create a simple test to ensure that the proper files are created and contain the right contents after running scaffold commands with the sample plugin. Unfortunately this sample doesn't scaffold anything that can actually be run on a cluster and just produces text files.